dev-utils/docker-compose.yml

```diff
@@ -74,46 +89,71 @@ services:
       test: ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--silent", "--output", "/dev/null", "http://kibana:5601/"]
       retries: 10
       interval: 10s
+    mem_limit: 2000m
+#    cpus: 0.5
     depends_on:
       elasticsearch:
         condition: service_healthy
     labels:
       - "co.elatic.apm.stack-version=${STACK_VERSION}"
+    networks:
+      apm:
+        aliases:
+          - kibana
 
-  apm-server-dev:
-    build:
-      context: ./apm-server-dev
-      args:
-        APM_SERVER_REPO: ${APM_SERVER_REPO}
-        APM_SERVER_BRANCH: ${APM_SERVER_BRANCH}
-    ports:
-      - "127.0.0.1:${APM_SERVER_PORT:-8200}:8200"
+  wait-for-apm-server:
```

Comment:
Do I understand it correctly that `wait-for-apm-server` only exists with 0 and does nothing else? if so the name is quite misleading.

Also are we doing this to make sure travis doesn't run the tests?

Reply:
this container is only to ensure that APM-Server is up before exit from `docker-compose up`, this ensures that you can run the test command with the resources needed, we use the option `-d` this means that we start the container in order but we do not wait for them to be ready, we only respect the dependency order because of that `wait-for-apm-server` will wait for APM Sevre to be healthy then `wait-for-apm-server` starts and exits.

` docker-compose -f dev-utils/docker-compose.yml up -d apm-server && docker ps `

```
WARNING: The SCOPE variable is not set. Defaulting to a blank string.
WARNING: The APM_SERVER_URL variable is not set. Defaulting to a blank string.
WARNING: The BUILD_NUMBER variable is not set. Defaulting to a blank string.
WARNING: The BRANCH_NAME variable is not set. Defaulting to a blank string.
WARNING: The SAUCE_USERNAME variable is not set. Defaulting to a blank string.
WARNING: The SAUCE_ACCESS_KEY variable is not set. Defaulting to a blank string.
WARNING: The MODE variable is not set. Defaulting to a blank string.
Starting elasticsearch ... done
Starting apm-server    ... done
CONTAINER ID        IMAGE                                                 COMMAND                  CREATED             STATUS                                     PORTS                                NAMES
324370bfb484        docker.elastic.co/apm/apm-server:7.1.0                "/usr/local/bin/dock…"   5 minutes ago       Up Less than a second (health: starting)   127.0.0.1:8200->8200/tcp             apm-server
8504ff34f279        docker.elastic.co/elasticsearch/elasticsearch:7.1.0   "/usr/local/bin/dock…"   11 minutes ago      Up 21 seconds (healthy)                    127.0.0.1:9200->9200/tcp, 9300/tcp   elasticsearch
23bc0ae8a7e8        portainer/portainer                                   "/portainer --admin-…"   27 hours ago        Up 27 hours                                0.0.0.0:9000->9000/tcp               portainerGUI

```